test/net/sourceforge/kolmafia/persistence/ConsumablesDatabaseTest.java

```diff
@@ -297,6 +302,16 @@ static void afterAll() {
       DebugDatabase.cacheItemDescriptionText(
           ItemPool.RING, html("request/test_normal_desc_item_ring.html"));
       TCRSDatabase.resetModifiers();
+      File root = KoLConstants.DATA_LOCATION;
+      String[] contents = root.list();
+      if (contents != null) {
```

Comment:
If you are going to keep this iteration, I would recommend a compare to null and return early rather than nest everything.

Reply:
Agreed, though the stream is better

Reply:
Right.  That interaction is gone but as discussed elsewhere I have an old, bad? habit from a time when one return per function was expected.  I don't think it was ALGOL W so it was probably FORTRAN with preprocessing so the coder could pretend there was some structure.  The preprocessor probably has not existed for several decades.